rclrs/src/node/mod.rs

```diff
@@ -312,3 +305,57 @@ impl Node {
         domain_id
     }
 }
+
+impl NodeBuilder {
+    /// Creates new builder for Node.
+    pub fn new(name: &str, context: &Context) -> NodeBuilder {
+        NodeBuilder {
+            context: context.handle.clone(),
+            name: name.to_string(),
```

Comment:
These member variables get converted from `&str` to `String` and then back to `&str` when building the `Node`, it'd be cleaner if either `&str` or `String` are used consistently everywhere.

Reply:
I'm leaning in the other direction. The cost is irrelevant here, and makes the API easier (not having to keep the owner of the `&str` alive, not having to type the lifetime parameter).

Reply:
It feels wrong that there are so many conversions back and forth. Instead, `&str` can be declared as `static` not avoid keeping the lifetimes.

Reply:
That would limit the API to not handle user-provided or randomly-generated node names.

I have the impulse to avoid conversions wherever possible as well – in most case it makes the code cleaner and smaller, but I think the impulse should be dismissed when it conflicts with a better API and performance is not a concern.

It's irrelevant from a practical perspective, but if it helps with this "feeling wrong", there is only one real conversion per field (the conversion from `String` to `&str` is free), and the `String` will not live forever and take up stack space, since the builder will be dropped pretty much immediately.

Reply:
Forgive me if this is a silly suggestion, but would a `cow` be of use in this instance?

Reply:
I think this struct will not affect performance because, in many case, this builder is used at initialization part of process lifetime. 

If @jhdcs's suggestion, using Cow, means for more user friendliness, I agree that and I think it us better to use impl Into<String>  instead, but if it means performance reason, it is excess I think.

Anyway, This should be discussed at another refactoring issue or PR. Especially, struct field is inner implementation and out of scope on user perspective.

Reply:
I think this struct is perfect as it is. Making the struct member and function parameter a `Cow` would avoid conversions, but the conversions are irrelevant and it would have similar ergonomics drawbacks (or worse) as a `&str` struct member. 

Making the function parameter an `impl Into<String>`, or `AsRef<str>` would allow also passing in `String` directly, but you can also just put an `&` before your String to make it coerce into a `&str`, so it doesn't buy us much. And in reality most people will pass in a string literal, so the extra complexity of the generic function signature would only be distracting to them.

Reply:
As for now, struct field types use current ones, not reference and others suggested here. It is the best to discuss in another issue dedicates for this topic, I think.